datalake/src/main/java/com/sequenceiq/datalake/flow/dr/restore/DatalakeRestoreActions.java

```diff
@@ -77,21 +121,30 @@ protected SdxContext createFlowContext(FlowParameters flowParameters, StateConte
             @Override
             protected void prepareExecution(DatalakeDatabaseRestoreStartEvent payload, Map<Object, Object> variables) {
                 super.prepareExecution(payload, variables);
-                variables.put(BACKUP_ID, payload.getBackupId());
+                if (!variables.containsKey(BACKUP_ID)) {
+                    variables.put(BACKUP_ID, payload.getBackupId());
+                }
                 if (!Strings.isNullOrEmpty(payload.getRestoreId())) {
                     variables.put(RESTORE_ID, payload.getRestoreId());
                 }
-                variables.put(OPERATION_ID, payload.getDrStatus().getOperationId());
+                if (!variables.containsKey(OPERATION_ID)) {
```

Comment:
why do you need to put RESTORE_ID and OPERATION_ID once again?

Reply:
Because we can have two entry points. if you go to database backup directly, you only set it once. 

Another sdx endpoint which backup the database only. So you can go to DATALAKE_DATABASE_RESTORE_START_STATE
without going to DATALAKE_TRIGGERING_RESTORE_STATE.

check through this.

    @Override
    @CheckPermissionByResourceName(action = AuthorizationResourceAction.RESTORE_DATALAKE)
    public SdxDatabaseRestoreResponse restoreDatabaseByName(@ResourceName String name, String backupId,
            String restoreId, String backupLocation) {